src/mds/CInode.cc

```diff
@@ -1628,6 +1638,10 @@ void CInode::encode_lock_state(int type, bufferlist& bl)
 	  dout(10) << fg << " " << dir->dirty_old_rstat << dendl;
 	  encode(fg, tmp);
 	  encode(dir->first, tmp);
+	  if (!is_auth()) {
+  	    encode(pf->rstat_dirty_from, tmp);
+	    pf->rstat_dirty_from = utime_t();
```

Comment:
it's too early to clear pf->rstat_dirty_from . pf->rstat_dirty_from  should be cleared by CInode::finish_scatter_update() or CDir::resync_accounted_rstat()

Reply:
@ukernel Hi, thanks for the review.
But it seems that clearing pf->rstat_dirty_from in CInode::finish_scatter_update() or CDir::resync_accounted_rstat() might errorly clear an rstat_dirty_from that's still needed. On the other hand, only non-auth MDSes who are trying to send their lock state to the auth MDS can reach the code that we are discussing now, and once pf->rstat_dirty_from is sent to the auth MDS, it should be free to clear. So I think maybe we should clear pf->rstat_dirty_from here. Am I right about this? Thanks:-)

Reply:
you need to consider the case that auth mds crash